assets/js/workflow-diagram/WorkflowDiagram.tsx

```diff
@@ -1,37 +1,33 @@
 import React, { useCallback, useEffect, useRef, useState } from 'react';
 import ReactFlow, {
   Controls,
-  ControlButton,
-  type NodeChange,
-  type ReactFlowInstance,
   ReactFlowProvider,
   applyNodeChanges,
   getRectOfNodes,
-  type Rect,
+  type NodeChange,
+  type ReactFlowInstance,
+  type Rect
 } from 'reactflow';
-import { useStore, type StoreApi } from 'zustand';
-import { shallow } from 'zustand/shallow';
 
+import { FIT_DURATION, FIT_PADDING } from './constants';
+import edgeTypes from './edges';
 import layout from './layout';
 import nodeTypes from './nodes';
-import edgeTypes from './edges';
-import usePlaceholders from './usePlaceholders';
 import useConnect from './useConnect';
+import usePlaceholders from './usePlaceholders';
 import fromWorkflow from './util/from-workflow';
+import shouldLayout from './util/should-layout';
 import throttle from './util/throttle';
 import updateSelectionStyles from './util/update-selection';
-import { FIT_DURATION, FIT_PADDING } from './constants';
-import shouldLayout from './util/should-layout';
 
-import type { WorkflowState } from '../workflow-editor/store';
+import { useWorkflowStore } from '../workflow-editor/store';
```

Comment:
As part of this work, we should move the store out to its own component. So the hook gets imported from `hooks/use-workflow-store` or something, and the component gets mounted directly the owning `.ex` file